pkg/sqlcmd/commands_test.go

```diff
@@ -226,6 +226,20 @@ func TestErrorCommand(t *testing.T) {
 	}
 }
 
+func TestOnErrorCommand(t *testing.T) {
+	s, buf := setupSqlCmdWithMemoryOutput(t)
+	defer buf.Close()
+	file, err := os.CreateTemp("", "sqlcmderr")
+	assert.NoError(t, err, "os.CreateTemp")
+	defer os.Remove(file.Name())
+
+	err = onerrorCommand(s, []string{""}, 1)
+	assert.EqualError(t, err, InvalidCommandError("ON ERROR", 1).Error(), "onerrorCommand with empty file name")
+
+	err = runSqlCmd(t, s, []string{":ONERROR ignore", "printtgit N'message'", "SELECT @@versionn", "GO"})
+	assert.NoError(t, err, "runSqlCmd")
```

Comment:
>assert.NoError(t, err, "runSqlCmd") [](http://example.com/codeflow?start=1&length=35)

you should also verify the error messages are in the output, otherwise it's possible nothing actually happened.

Reply:
Incorporated